Automattic/sensei
# Add Take Course block

### Changes proposed in this Pull Request

* Add Take Course block
* Add PHP render callback, wrapping block content in a form that starts the course for the user
* Link to Login/Register (on My Courses page if it exists) if user is not logged in — using same logic as the original
* Disable button with a note if there is a prerequisite course

### Parity with non-block version

- [x] Handle course prerequisite missing
- [x] Only display if user can enroll
- [x] Register/login messaging
- ~Add filter to allow overriding it~ 
- ~WCPC takeover when course is paid~ *To be explored in purchase button PRs / needs further discussion*

### Testing instructions

* Add a take course block in the editor, save & publish
* Open the course with a signed out user
* Take course button should appear and take the user to the login page
--
* Open as a logged in, not enrolled user
* Take course button should appear and enroll the user upon clicking
* After enrolling, Take course button should not appar
-- 
* Add a prerequisite for the course
* Open as logged in, not enrolled user
* Take course button should be disabled with a notice about prerequisite course


### New/Updated Hooks

Now also calling these filters in the block:
 
* sensei_course_complete_prerequisite_message
* sensei_use_wp_register_link



### Screenshot / Video

<img width="640" alt="image" src="https://user-images.githubusercontent.com/176949/96051271-71538f80-0e7b-11eb-875c-79634990a57d.png">


#### Disabled with prerequisite:

<img width="617" alt="image" src="https://user-images.githubusercontent.com/176949/96051155-39e4e300-0e7b-11eb-84ba-45a662a2f81a.png">
